### PR TITLE
Changes to update Power VS PlatformSpec with ServiceEndpoints

### PR DESCRIPTION
**Pull request by Karthik-K-N**
This PR contains the changes to update PowerVSPlatformSpec with the service endpoints. This will help in overriding the default services with the user provided custom service endpoitns.

[cluster-config-operator](https://github.com/openshift/cluster-config-operator/) is responsible for validating, setting the defaults if required and finally updating results to the status field for other operators to consume the information. service endpoints under `spec` field and updates the `status` filed

Here is example for AWS in cluster-config-operator: https://github.com/openshift/cluster-config-operator/blob/9fe03c98571bd0bb311ada3147bb6a091b114ddc/pkg/operator/aws_platform_service_location/controller.go#L79-L94

Here is example how its being consumed in ingress-operator: https://github.com/openshift/cluster-ingress-operator/blob/8a45188b18ec8122359449b9df9c38960595854d/pkg/operator/controller/dns/controller.go#L488

Related to enhancement: openshift/enhancements#736

### CODE REVIEW

File: config/v1/types_infrastructure.go [hunk 1]
```diff
@@ -617,7 +617,12 @@ type PowerVSServiceEndpoint struct {
 
 // PowerVSPlatformSpec holds the desired state of the IBM Power Systems Virtual Servers infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type PowerVSPlatformSpec struct{}
+type PowerVSPlatformSpec struct {
+	// serviceEndpoints is a list of custom endpoints which will override the default
+	// service endpoints of a Power VS service.
+	// +optional
```
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply config/v1/types_infrastructure.go:629]: Added the changes in the recent commit

File: config/v1/0000_10_config-operator_01_infrastructure.crd.yaml [hunk 2]
```diff
@@ -98,6 +98,26 @@ spec:
...
+                          type: array
+                          items:
+                            description: PowervsServiceEndpoint stores the configuration of a custom url to override existing defaults of PowerVS Services.
+                            type: object
+                            required:
+                              - name
+                              - url
+                            properties:
+                              name:
+                                description: name is the name of the Power VS service.
```
[the remark this answers is not in the document]
Comment (Karthik-K-N) [reply config/v1/0000_10_config-operator_01_infrastructure.crd.yaml]: Added few sample services like iam, power endpoint.

**Comment by Karthik-K-N** [comment]
All the review comments has been addressed

**Comment by sttts** [comment]
@wking can you lgtm? Or whoever is owning the infrastructure resource.

Looks fine from my side. Will approve.

**Comment by wking** [comment]
I don't think I'm an owner of the Infrastructure resource, but "doing the same thing AWS has been doing, but now for PowerVS" sounds sane to me :+1:

**Review by wking — APPROVED** [review]

**Comment by sttts** [comment]
@Karthik-K-N in your description you write
> cluster-config-operator is responsible for validating, setting the defaults if required and finally updating results to the status field for other operators to consume the information. service endpoints under spec field and updates the status filed

Not sure what the purpose is of the statement, but I don't agree that's the task of cluster-config-operator. Its only reason for existence is to carry the config CRDs to decouple them from the consuming operators. But I don't think cluster-config-operator should ever be responsible to report status. It's even questionable to have status in config CRDs.

**Comment by Karthik-K-N** [comment]
> @Karthik-K-N in your description you write
> 
> > cluster-config-operator is responsible for validating, setting the defaults if required and finally updating results to the status field for other operators to consume the information. service endpoints under spec field and updates the status filed
> 
> Not sure what the purpose is of the statement, but I don't agree that's the task of cluster-config-operator. Its only reason for existence is to carry the config CRDs to decouple them from the consuming operators. But I don't think cluster-config-operator should ever be responsible to report status. It's even questionable to have status in config CRDs.

So ya, The reason i have mentioned about cluster config operator is because as an example how the aws provider handled it.
Some reference: https://github.com/openshift/enhancements/pull/163/files#diff-2c505260612022d64fdf1a68576c2e0027a1bd9e1ccfef945e6120790ac3009aR191
Cluster config operator: https://github.com/openshift/cluster-config-operator/blob/9fe03c98571bd0bb311ada3147bb6a091b114ddc/pkg/operator/aws_platform_service_location/controller.go#L79-L94

**Comment by sttts** [comment]
There is no clear owner of the infra resource it seems.

/lgtm
/approve

**Pull request merged**